### PR TITLE
[Performance Optimization] Improve the performance of tf.clip_by_norm when the input is IndexedSlices

### PR DESCRIPTION
**Pull request by wangsiyu**
Currently, the `tf.clip_by_norm` will convert the input to dense tensor and do the clipping. However, the gradients may be `IndexedSlices` type when encounters with `tf.nn.embedding_lookup` in forward pass. It may both consume large memory and make computation performance worse.

In this pull request, the clipping function will deal with `IndexedSlices` directly instead of converting to dense tensor.

### CODE REVIEW

**Comment by wangsiyu** [comment]
Any code review comments are welcome. Thanks. @jhseu @alextp

**Review by rmlarsen — DISMISSED** [review]

**Comment by rmlarsen** [comment]
@wangsiyu please address the linter errors here:
https://source.cloud.google.com/results/invocations/a7086d3b-2b1e-448f-8c1d-37282e599faf/log

**Comment by wangsiyu** [comment]
> @wangsiyu please address the linter errors here:
> https://source.cloud.google.com/results/invocations/a7086d3b-2b1e-448f-8c1d-37282e599faf/log

I have refine the code according to the pylint feedback. Please review again. Thanks

**Comment by wangsiyu** [comment]
Any update for me？

**Comment by wt-huang** [comment]
@wangsiyu It seems there are still a few pylint issues. Please check.

**Comment by wangsiyu** [comment]
@wt-huang I'm sorry. Could you show me more details? The lint check of `clip_ops.py` and `clip_ops_test.py` seems ok locally.  Thanks very much.

**Comment by wangsiyu** [comment]
@wt-huang Ci build test shows all lint passed.

**Comment by wt-huang** [comment]
@wangsiyu good that all pylint issues are fixed. Thanks!

**Comment by wangsiyu** [comment]
@wt-huang Thanks very much. But lack of an approval here @rmlarsen .
